### PR TITLE
add safety limits for subaru outback/legacy

### PR DESCRIPTION
**Pull request by bugsy924**

### CODE REVIEW

File: board/safety/safety_subaru.h [hunk 1]
```diff
@@ -40,19 +45,34 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
+    desired_torque = ((to_send->RDLR >> 16) & 0x1FFF);
+    desired_torque = to_signed(desired_torque, 13);
+    } else if (addr == 0x164) {
+    desired_torque = ((to_send->RDLR >> 8) & 0x1FFF);
+    desired_torque = to_signed(desired_torque, 13);
+    }
+    
     int violation = 0;
     uint32_t ts = TIM2->CNT;
-    desired_torque = to_signed(desired_torque, 13);
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: This can stay here. Same code in both cases

Comment (bugsy924) [reply board/safety/safety_subaru.h]: I'll fix it now

File: board/safety/safety_subaru.h [hunk 2]
```diff
@@ -40,15 +45,29 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
 static int subaru_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
   uint32_t addr = to_send->RIR >> 21;
 
   // steer cmd checks
-  if (addr == 0x122) {
-    int desired_torque = ((to_send->RDLR >> 16) & 0x1FFF);
+  if (addr == 0x122 || addr == 0x164) {
+    int desired_torque = 0;
+    if (addr == 0x122) {
+    desired_torque = ((to_send->RDLR >> 16) & 0x1FFF);
```
Comment (dekerr) [inline board/safety/safety_subaru.h]: i don't want to nag, but would you mind indenting desired_torque? :)

File: board/safety/safety_subaru.h [hunk 3]
```diff
@@ -40,15 +45,29 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
+    } else if (addr == 0x164) {
+    desired_torque = ((to_send->RDLR >> 8) & 0x1FFF);
```
Comment (legonigel) [inline board/safety/safety_subaru.h]: Another indentation to fix here.

File: board/safety/safety_subaru.h [hunk 4]
```diff
@@ -40,15 +45,29 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
       controls_allowed = 0;
     }
     subaru_cruise_engaged_last = cruise_engaged;
+  } else if ((addr == 0x144) && (bus_number == 0)) {
+    int cruise_engaged = (to_push->RDHR >> 17) & 1;
+    if (cruise_engaged && !subaru_cruise_engaged_last) {
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: this should also be combined with ^^. 
Let me refactor it. I'll open a new PR. @bugsy924

File: board/safety/safety_subaru.h [hunk 5]
```diff
@@ -29,6 +29,11 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
     torque_driver_new = to_signed(torque_driver_new, 11);
     // update array of samples
     update_sample(&subaru_torque_driver, torque_driver_new);
+  } else if ((addr == 0x371) && (bus_number == 0)){
+    int torque_driver_new = ((to_push->RDLR >> 29) & 0x7FF);
+    torque_driver_new = to_signed(torque_driver_new, 10);
```
Comment (rbiasini) [inline board/safety/safety_subaru.h]: 10 or 11 bits? 0x7FF is 11.

**Comment by rbiasini** [comment]
See https://github.com/commaai/panda/pull/189

**Pull request closed without merging**